### ISSUE TITLE
RUM Data Not Recorded for WebView on Android

### ISSUE THREAD

**Issue by tksou** [DataDog/dd-sdk-flutter#670]
### Question
I am trying to use the example provided at https://github.com/DataDog/dd-sdk-flutter/tree/develop/packages/datadog_webview_tracking/example to record RUM data for WebView using Datadog. The recording works correctly on iOS, but it does not record anything on Android. Additionally, the following logs are displayed in the console during execution. This issue occurs on both emulators and physical devices.

### Logs

```
E/Datadog ( 7273): [_dd.sdk_core.default]: Uncaught exception during the task execution
E/Datadog ( 7273): java.lang.NoSuchMethodError: No interface method write(Lcom/datadog/android/api/storage/EventBatchWriter;Ljava/lang/Object;)Z in class Lcom/datadog/android/api/storage/DataWriter; or its super classes (declaration of 'com.datadog.android.api.storage.DataWriter' appears in /data/app/~~XRDqsLm8VFd9s5KRvGrVLw==/com.datadoghq.flutter.webview.example-yQdeegDCeqn6nwTtReGrGw==/base.apk)
E/Datadog ( 7273):      at com.datadog.android.webview.internal.rum.WebViewRumEventConsumer$consume$1.invoke(WebViewRumEventConsumer.kt:46)
E/Datadog ( 7273):      at com.datadog.android.webview.internal.rum.WebViewRumEventConsumer$consume$1.invoke(WebViewRumEventConsumer.kt:41)
E/Datadog ( 7273):      at com.datadog.android.core.internal.SdkFeature$withWriteContext$1.invoke(SdkFeature.kt:177)
E/Datadog ( 7273):      at com.datadog.android.core.internal.SdkFeature$withWriteContext$1.invoke(SdkFeature.kt:177)
E/Datadog ( 7273):      at com.datadog.android.core.internal.persistence.ConsentAwareStorage.writeCurrentBatch$lambda$1(ConsentAwareStorage.kt:89)
E/Datadog ( 7273):      at com.datadog.android.core.internal.persistence.ConsentAwareStorage.$r8$lambda$SRgNYV2h4JQ8NXSPfSTUU6FujmQ(Unknown Source:0)
E/Datadog ( 7273):      at com.datadog.android.core.internal.persistence.ConsentAwareStorage$$ExternalSyntheticLambda0.run(Unknown Source:10)
E/Datadog ( 7273):      at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
E/Datadog ( 7273):      at java.util.concurrent.FutureTask.run(FutureTask.java:264)
E/Datadog ( 7273):      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
E/Datadog ( 7273):      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
E/Datadog ( 7273):      at java.lang.Thread.run(Thread.java:1012)
```

### Steps to Reproduce
1. Clone the repository and navigate to the example directory: https://github.com/DataDog/dd-sdk-flutter/tree/develop/packages/datadog_webview_tracking/example
2. Follow the instructions to set up the example for both iOS and Android.
3. Ensure the following package versions are used:
   - datadog_flutter_plugin: ^2.8.0
   - datadog_webview_tracking: ^2.0.2
   - webview_flutter: ^4.8.0
4. Use Flutter version 3.19.0.
5. Run the example on an Android device.
6. Observe that RUM data is not recorded on Datadog for Android, while it works correctly for iOS.
7. Check the console logs on Android for the error messages mentioned above.

### Expected Behavior
RUM data should be recorded for WebView on both iOS and Android.

#### Android
<img src="https://github.com/user-attachments/assets/bce6c718-7255-4b41-bf92-fcdb166b9bfe" width=70%>

#### iOS
<img src="https://github.com/user-attachments/assets/c7ca0413-03f8-4524-b63e-654ed28c0f1f" width=70%>

### Environment

<details>
<summary>flutter doctor -v</summary>

```
$ flutter doctor -v
[✓] Flutter (Channel stable, 3.19.0, on macOS 14.2.1 23C71 darwin-arm64, locale ja-JP)
    • Flutter version 3.19.0 on channel stable at /Users/user/fvm/versions/3.19.0
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision bae5e49bc2 (9 months ago), 2024-02-13 17:46:18 -0800
    • Engine revision 04817c99c9
    • Dart version 3.3.0
    • DevTools version 2.31.1

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/user/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio 2.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio 2.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[✓] VS Code (version 1.95.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.100.0

[✓] Connected device (5 available)

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```
</details>

Please let me know if there is any additional information or steps I can provide to help diagnose this issue. Thank you!

**Comment by fuzzybinary** [comment]
Hi @tksou ,

Thanks for reporting!

It looks like we may have introduced an incompatibility between the main package and the webview package because of a change on the native side. I'll try to get this fixed and release an updated version of this package next week.

If you need an immediate workaround (and want to check if this is the problem) you can force your application to take the updated native dependency by adding this to your `app/build.gradle`:

```groovy
dependencies {
    implementation "com.datadoghq:dd-sdk-android-webview:2.14.0"
}
```

If you already have a dependencies section just add the one line.

**Comment by tksou** [comment]
Hi @fuzzybinary ,

Thank you very much for your prompt response.

As you suggested, adding the dependency resolved the issue and RUM data is now being recorded from the WebView on Android. Your assistance was very helpful in identifying the cause of the problem.

> I'll try to get this fixed and release an updated version of this package next week.

Understood. I look forward to the package update.
Thanks again for your support!

**Comment by fuzzybinary** [comment]
Thanks @tksou!

Good to hear we have a workaround for you. I'm going to reopen though as this is still an issue that needs to be resolved. I'll close it when I release the new version.

**Comment by fuzzybinary** [comment]
Hi @tksou 

This should now be fixed in `datadog_webview_tracking` version 2.1.0 which is now on pub.

Thanks for reporting and sorry for the inconvenience.

**Comment by tksou** [comment]
Hi @fuzzybinary ,

Thank you! I have tried it with datadog_webview_tracking version 2.1.0, and it works perfectly even after removing the workaround.
Thanks again for the quick fix!
